test/framework/src/main/java/org/opensearch/test/OpenSearchTestClusterRule.java

```diff
@@ -226,8 +233,8 @@ private static boolean isSuiteScopedTest(Class<?> clazz) {
         return clazz.getAnnotation(SuiteScopeTestCase.class) != null;
     }
 
-    private boolean hasParametersChanged(final ParameterizedOpenSearchIntegTestCase target) {
-        return !((ParameterizedOpenSearchIntegTestCase) suiteInstance).hasSameParametersAs(target);
+    private static boolean hasParametersChanged(final OpenSearchIntegTestCase instance, final ParameterizedOpenSearchIntegTestCase target) {
+        return !((ParameterizedOpenSearchIntegTestCase) instance).hasSameParametersAs(target);
```

Comment:
Nitpick, can you change the argument type to avoid the cast? It looks like the type is checked prior to every call of this method.

Also, I think `haveParametersChanged()` is grammatically correct (can't believe I missed this pedantry on the previous PR)

Reply:
I was thinking about `hasTestClassParametersChanged`, should be correct?